microprofile/server/src/main/java/io/helidon/microprofile/server/ServerCdiExtension.java

```diff
@@ -416,31 +448,40 @@ private static int priority(Class<?> aClass) {
     private void registerWebServerService(Class<?> serviceClass,
                                           Service service) {
 
-        RoutingPath rp = serviceClass.getAnnotation(RoutingPath.class);
-        RoutingName rn = serviceClass.getAnnotation(RoutingName.class);
+        RoutingPath rp;
+        RoutingName rn;
+        Config serviceConfig;
+
+        if (producedServices.containsKey(service)) {
+            // created with producer
+            AnnotatedMember<?> producerAnnotatedMember = producedServices.get(service);
+            rp = producerAnnotatedMember.getAnnotation(RoutingPath.class);
+            rn = producerAnnotatedMember.getAnnotation(RoutingName.class);
+            Member member = producerAnnotatedMember.getJavaMember();
+            serviceConfig = config.get(member.getDeclaringClass().getName() + "." + member.getName());
+        } else {
+            // standalone bean
+            rp = serviceClass.getAnnotation(RoutingPath.class);
```

Comment:
This isn't guaranteed to work in the presence of portable extensions.  You must always use an `Annotated` to get annotations, since extensions may add or remove extensions.

If you have a "regular" `Class`, you can always "get" an `AnnotatedClass` representing it by using `BeanManager::createAnnotatedType`.

Reply:
Thx, annotations age grabbed over `getAnnotated` now
https://github.com/oracle/helidon/blob/320be29cc2c9779f53a10a8de7bd64e61fb6cfc4/microprofile/server/src/main/java/io/helidon/microprofile/server/RoutingConfiguration.java#L29-L33